src/content/components/TreeView.js

```diff
@@ -119,16 +120,25 @@ class TreeViewRowScrolledColumns extends Component {
     }
   }
 
+  _renderIcon(iconUrl) {
+    return <img
+            src={iconUrl}
+            referrerPolicy='no-referrer'
+            className='treeRowIcon'
+            onError={ e => this.props.onIconError(this.props.nodeId, e.target.src) } />;
+  }
+
   render() {
     const {
       node, depth, mainColumn, appendageColumn, index, canBeExpanded,
-      isExpanded, selected, highlightString, appendageButtons,
+      isExpanded, selected, highlightString, appendageButtons, icon,
     } = this.props;
     const evenOddClassName = (index % 2) === 0 ? 'even' : 'odd';
     return (
       <div className={`treeViewRow treeViewRowScrolledColumns ${evenOddClassName} ${selected ? 'selected' : ''} ${node.dim ? 'dim' : ''}`} style={{height: '16px'}} onClick={this._onClick}>
         <span className='treeRowIndentSpacer' style={{ width: `${depth * 10}px` }}/>
         <span className={`treeRowToggleButton ${isExpanded ? 'expanded' : 'collapsed'} ${canBeExpanded ? 'canBeExpanded' : 'leaf'}`} />
+        { icon && this._renderIcon(icon) }
```

Comment:
Sorry my comment was a little out of the blue! Familiarity is a big part of it, not to bikeshed on it too much as it was a passing comment, but I think a ternary is more explicit as there is a clear condition and return values, while a boolean's first operand combines the condition and the returning value, which is more work to understand. I would say this falls into a similar category as the rule for not doing assignment in a conditional. It works but is a bit magical.